internal/mode/static/state/change_processor.go

```diff
@@ -158,6 +159,11 @@ func NewChangeProcessorImpl(cfg ChangeProcessorConfig) *ChangeProcessorImpl {
 				store:             newObjectStoreMapAdapter(clusterStore.Secrets),
 				trackUpsertDelete: false,
 			},
+			{
+				gvk:               extractGVK(&ngfAPI.NginxProxy{}),
```

Comment:
do we need to track relationships? so that we only regenerate the graph if the referenced NginxProxy resource is upserted, etc?

Reply:
Our predicate ensures that we only get updates for the referenced NginxProxy.